src/components/pages/VisualisationEditor.tsx

```diff
@@ -33,15 +37,19 @@ const VisualisationEditor: React.FC = () => {
         return <Text>Loading…</Text>;
     }
 
-    if (!data || !data.allMetadata.some((el) => el.id === id)) {
+    if (typeof data === 'undefined' || ((!data || !data.allMetadata.some((el) => el.id === id)) && !visualisation)) {
```

Comment:
Why is `typeof data === 'undefined'` used instead of just `!data` or `data === undefined`?

Reply:
This `if`-statement would also benefit from being broken up into smaller, more legible pieces. Right now it's a bit difficult to understand.

Reply:
Because typescript does not accept it 🗡️

Reply:
![image](https://user-images.githubusercontent.com/43110067/115541367-5422d200-a29f-11eb-97f3-d9a7b35e932e.png)

Reply:
Then `data === undefined` would be a cleaner check.

Generally, using `typeof xxx === ` should be avoided unless you're working with custom classes and need to check for instances of them.

Reply:
Typeof is rather strange with quirks like

```
// This stands since the beginning of JavaScript
typeof null === 'object';
```

Reply:
If it cannot be null, then an explicit check for `data === undefined` should suffice.